lib/DxcSupport/HLSLOptions.cpp

```diff
@@ -208,7 +208,8 @@ int ReadDxcOpts(const OptTable *optionTable, unsigned flagsToInclude,
   DXASSERT(opts.ExternalLib.empty() == opts.ExternalFn.empty(),
            "else flow above is incorrect");
 
-  opts.OutputWarnings = Args.hasFlag(OPT_no_warnings, OPT_INVALID, true);
+  // when no-warnings option is present, do not output warnings.
+  opts.OutputWarnings = !Args.hasFlag(OPT_no_warnings, OPT_INVALID, false);
```

Comment:
OPT_no_warnings is really a negative option, so I think we it would be better to write this like:
    
    opts.OutputWarnings = Args.hasFlag(OPT_INVALID, OPT_no_warnings, true);

That is, by default warnings are enabled but the OPT_no_warnings will turn them off.